src/audio/component.c

```diff
@@ -326,121 +326,210 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 	return 0;
 }
 
-void comp_free_model_data(struct comp_dev *dev, struct comp_model_data *model)
+/** \brief Struct handler for large component configs */
+struct comp_data_blob_handler {
+	uint32_t data_size;	/**< size of component's model data */
+	void *data;		/**< pointer to model data */
+	void *data_new;		/**< pointer to model data */
+	bool data_ready;	/**< set when fully received */
```

Comment:
data and data_new have the same description. This is confusing.

Reply:
@dbaluta 
I've updated it.